### PR TITLE
fix: Move settings tabs to common file

### PR DESCRIPTION
**Pull request by tsa96**
Both the main settings and the search js files use this to I moved it to a separate file. Fixes a JS error that occurs when trying to search settings.

### CODE REVIEW

File: layout/mainmenu_settings.xml [hunk 1]
```diff
@@ -4,6 +4,7 @@
 	</styles>
 
 	<scripts>
+		<include src="file://{resources}/scripts/common/settings.js" />
 		<include src="file://{resources}/scripts/settings_shared.js" />
```
Comment (Gocnak) [inline layout/mainmenu_settings.xml:8]: Should settings_shared stuff be moved to common/settings.js ?

Comment (tsa96) [reply layout/mainmenu_settings.xml:8]: Mmh no, settings_shared is actually a bunch of unique functionality for different settings pages, the structure is a carryover from CS:GO's settings. I'd quite like to refactor that but too much work for little gain currently IMO.

File: scripts/common/settings.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
+const SETTINGS_TABS = {
```
Comment (Gocnak) [inline scripts/common/settings.js:1]: Missing `'use strict';`
```suggestion
'use strict';

const SETTINGS_TABS = {
```

**Review by Gocnak — CHANGES_REQUESTED** [review]

File: scripts/common/settings.js (hunk 2 above)
Comment (tsa96) [reply scripts/common/settings.js:3]: Fixed

**Review by Gocnak — APPROVED** [review]

**Pull request merged**